### ISSUE TITLE
Proposal: Support config time zone for log driver message

### ISSUE THREAD

**Issue by hustcat** [moby/moby#33778]
[Current log driver message](https://github.com/moby/moby/blob/master/daemon/logger/copier.go#L102) is hard coded with `UTC` time, how about add `--log-timezone` for `log driver` to config message time zone?

**Comment by juliohm1978** [comment]
This is an interesting proposal. This would make deploying log centralization solutions, such as Graylog, easier. While collectors can easily collect from all containers, there is a manual overhead configuring timezones on each and every container image that is deployed.

Suffice to say that official images for popular stacks have no consensus on how to provide the timezone to containers; each will have their own way of doing this (TZ env var, /etc/timezone, custom config, etc). Managing all of them individually can become messy.

A log aggregator can easily ignore timestamps from applications inside the container and keep only the timestamp from docker log files. However, we are unable to elegantly implement this if dockerd itself is cannot to use another timezone, other than UTC.

An expected beahvior from dockerd would be to use the host's timezone as a default and allow customization via command line parameters or its config files.

**Comment by vieux** [comment]
I think that's a good idea.

**Comment by ripcurld0** [comment]
calling #dibs

**Comment by ripcurld0** [comment]
I am closing this as discussed in https://github.com/moby/moby/pull/35051#issuecomment-334117784 and moving it to https://github.com/docker/cli/issues/604.

**Comment by iFredOS** [comment]
Still not fixed!

**Comment by Z18Aplha** [comment]
just if somebody is interested in: still not fixed, using on DSM (Synology).
I cant imagine this fix is a huge problem..

**Comment by thaJeztah** [comment]
There's some discussion in https://github.com/moby/moby/pull/35051#issuecomment-334117784 - consensus was to (if implemented) do this on the client side, so when presenting the logs, instead of daemon side.

**Comment by ericxob77** [comment]
Yes, it is very irritating that I can't chang the timestamp in docker logs!  Please make it honor TZ value in environment variables.

**Comment by prax93** [comment]
Still not working. The Environment Variable TZ gets ignored. However inside the Container the Variable is set and the date command prints the correct date out.
This is a issue when Monitoring Containers and following Log Entires accordingly.
Also time relevant in App Cronjobs will be difficult to Monitor.
